libcontainer/cgroups/fs/cpuset.go

```diff
@@ -48,7 +50,109 @@ func (s *CpusetGroup) Remove(d *cgroupData) error {
 	return removePath(d.path("cpuset"))
 }
 
+func getNumericStat(path string, filename string, extracted *[]uint16) error {
+	fileContent, err := fscommon.GetCgroupParamString(path, filename)
+	if err != nil {
+		return err
+	}
+	if len(fileContent) == 0 {
+		return fmt.Errorf("%s found to be empty", filepath.Join(path, filename))
+	}
+
+	for _, s := range strings.Split(fileContent, ",") {
+		if strings.Contains(s, "-") {
+			splitted := strings.Split(s, "-")
+			if len(splitted) != 2 {
+				return fmt.Errorf("Couldn't parse %s", filepath.Join(path, filename))
+			}
+			min, err := strconv.ParseUint(splitted[0], 10, 16)
+			if err != nil {
+				return err
+			}
+			max, err := strconv.ParseUint(splitted[1], 10, 16)
+			if err != nil {
+				return err
+			}
+			if min > max {
+				return fmt.Errorf("Couldn't parse %s", filepath.Join(path, filename))
+			}
+			for i := min; i <= max; i++ {
+				*extracted = append(*extracted, uint16(i))
+			}
+		} else {
+			value, err := strconv.ParseUint(s, 10, 16)
+			if err != nil {
+				return err
+			}
+			*extracted = append(*extracted, uint16(value))
+		}
+
+	}
+
+	return nil
+}
+
 func (s *CpusetGroup) GetStats(path string, stats *cgroups.Stats) error {
+	var (
+		err  error
+		cpus []uint16
+		mems []uint16
+	)
+	err = getNumericStat(path, "cpuset.cpus", &cpus)
+	if err != nil {
+		return err
+	}
+	cpuExclusive, err := fscommon.GetCgroupParamUint(path, "cpuset.cpu_exclusive")
+	if err != nil {
+		return err
+	}
+	err = getNumericStat(path, "cpuset.mems", &mems)
+	if err != nil {
+		return err
+	}
+	memHardwall, err := fscommon.GetCgroupParamUint(path, "cpuset.mem_hardwall")
+	if err != nil {
+		return err
+	}
+	memExclusive, err := fscommon.GetCgroupParamUint(path, "cpuset.mem_exclusive")
+	if err != nil {
+		return err
+	}
+	memoryMigrate, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_migrate")
+	if err != nil {
+		return err
+	}
+	memorySpreadPage, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_spread_page")
+	if err != nil {
+		return err
+	}
+	memorySpreadSlab, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_spread_slab")
+	if err != nil {
+		return err
+	}
+	memoryPressure, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_pressure")
+	if err != nil {
+		return err
+	}
+	schedLoadBalance, err := fscommon.GetCgroupParamUint(path, "cpuset.sched_load_balance")
+	if err != nil {
+		return err
+	}
+	schedRelaxDomainLevel, err := fscommon.GetCgroupParamInt(path, "cpuset.sched_relax_domain_level")
```

Comment:
Are you sure you want to return an error if any of these files do not exist? What if some older kernel do not provide some?

Reply:
Changed stat gathering method to try to gather as much as possible instead of quitting at first error